watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2_P_extension.py

```diff
@@ -658,6 +678,184 @@ def solve(m, solver=None):
+        costing_method=cost_primary_clarifier,
+    )
+
+    m.fs.CL2.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=cost_circular_clarifier,
+    )
+
+    m.fs.AD.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.dewater.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.thickener.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+
+    # TODO: Leaving out mixer costs; consider including later
+
+    # process costing and add system level metrics
+    m.fs.costing.cost_process()
+    m.fs.costing.add_electricity_intensity(m.fs.FeedWater.properties[0].flow_vol)
+    m.fs.costing.add_annual_water_production(m.fs.Treated.properties[0].flow_vol)
+    m.fs.costing.add_LCOW(m.fs.FeedWater.properties[0].flow_vol)
+    m.fs.costing.add_specific_energy_consumption(m.fs.FeedWater.properties[0].flow_vol)
+
+    m.fs.objective = pyo.Objective(expr=m.fs.costing.LCOW)
+    iscale.set_scaling_factor(m.fs.costing.LCOW, 1e3)
+    iscale.set_scaling_factor(m.fs.costing.total_capital_cost, 1e-7)
+    iscale.set_scaling_factor(m.fs.costing.total_capital_cost, 1e-5)
+
+    iscale.calculate_scaling_factors(m.fs)
+
+
+def display_costing(m):
+    print("Levelized cost of water: %.2f $/m3" % pyo.value(m.fs.costing.LCOW))
+
+    print(
+        "Total operating cost: %.2f $/yr" % pyo.value(m.fs.costing.total_operating_cost)
+    )
+    print("Total capital cost: %.2f $" % pyo.value(m.fs.costing.total_capital_cost))
+
+    print(
+        "Total annualized cost: %.2f $/yr"
+        % pyo.value(m.fs.costing.total_annualized_cost)
+    )
+    print(
+        "Specific energy consumption with respect to influent flowrate: %.1f kWh/m3"
+        % pyo.value(m.fs.costing.specific_energy_consumption)
+    )
+
+    print(
+        "electricity consumption R5",
+        pyo.value(m.fs.R5.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.R5.electricity_consumption[0]),
+    )
+    print(
+        "electricity consumption R6",
+        pyo.value(m.fs.R6.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.R6.electricity_consumption[0]),
+    )
+    print(
+        "electricity consumption R7",
+        pyo.value(m.fs.R7.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.R7.electricity_consumption[0]),
+    )
+    print(
+        "electricity consumption primary clarifier",
+        pyo.value(m.fs.CL.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.CL.electricity_consumption[0]),
+    )
+    print(
+        "electricity consumption secondary clarifier",
+        pyo.value(m.fs.CL2.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.CL2.electricity_consumption[0]),
+    )
+    print(
+        "electricity consumption AD",
+        pyo.value(m.fs.AD.electricity_consumption[0]),
+        pyo.units.get_units(m.fs.AD.electricity_consumption[0]),
+    )
+    print(
```

Comment:
Looks like a chunk of these printouts aren't technically under the "costing" category. Maybe you can consider moving those under a separate display function.

Reply:
Sure, I'll address your comments for the conventional and extended BSM2 flowsheets